### PR TITLE
Add initial support for rule profiles

### PR DESCRIPTION
**Pull request by ssbarnea**
This change is advertised as experimental because it includes listing of multiple planned rules which we will implement over the following months. They are documented here in order to set expectations about at which profile level users should expect to see them.

This feature does not change the behavior of ansible-lint unless user is adding `--profile` argument.

Fixes: #2109

### CODE REVIEW

**Review by cidrblock — APPROVED** [review]

File: src/ansiblelint/cli.py [hunk 1]
```diff
@@ -258,6 +258,18 @@ def get_cli_parser() -> argparse.ArgumentParser:
         action="count",
         help="quieter, reduce verbosity, can be specified twice.",
     )
+    parser.add_argument(
+        "-P",
+        "--profile",
+        # * allow to distinguish between calling without -P, with -P and
+        # with -P=foo, while '?' does not. We do not support a real list.
+        nargs="*",
```
[the remark this answers is not in the document]
Comment (ssbarnea) [reply src/ansiblelint/cli.py:266]: It does not work like this, `*` allows you to repeat the command lie `-P foo -P bar` but not doing something like `-P foo bar`. In fact we do support only one argument but we are forced to use `*` due to a limitation in argparse.

File: src/ansiblelint/__main__.py [hunk 2]
```diff
@@ -170,6 +170,17 @@ def main(argv: Optional[List[str]] = None) -> int:
...
 
+    if options.profile == []:
+        from ansiblelint.generate_docs import profiles_as_rich
+
+        console.print(profiles_as_rich())
+        return 0
+    if options.profile:
+        from ansiblelint.rules import filter_rules_with_profile
+
+        filter_rules_with_profile(rules, options.profile[0])
```
[the remark this answers is not in the document]
Comment (ssbarnea) [reply src/ansiblelint/__main__.py:181]: Good point, I will add an error if len>1, as we do not support it.

[the remark this answers is not in the document]
Comment (ssbarnea) [reply src/ansiblelint/__main__.py:181]: Apparently I do not need to do anything as argparse will only return one item, even if I mention the command multiple time, the last one. I was not able to make it return a list with more than one entry.

File: src/ansiblelint/data/profiles.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    risky-file-permissions:
+    risky-octal:
+    risky-shell-pipe:
+shared:
+  extends: safety
+  description: >
+    The shared profile is for those that want to publish their content, roles
+    or collection to either [galaxy](https://galaxy.ansible.com) or to another
+    automation-hub instance. In addition to all the rules related to packaging
+    and publishin like metadata checks, this also includes some rules that are
```
Comment (ganeshrn) [inline src/ansiblelint/data/profiles.yml:80]: s/publishin/publishing

File: src/ansiblelint/data/profiles.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    max-tasks:
+      url: https://github.com/ansible/ansible-lint/issues/2172
+    unsafe-loop:
+      # unsafe-loop[prefix] (currently named "no-var-prefix")
+      # [unsafe-loop[var-prefix|iterator]]
+      url: https://github.com/ansible/ansible-lint/issues/2038
+production:
+  description: >
+    The production profile represents the top level of verification and it
+    is required for inclusion in AAP as either validated or certified content.
```
Comment (ganeshrn) [inline src/ansiblelint/data/profiles.yml:110]: AAP full form can be mentioned here

File: src/ansiblelint/generate_docs.py [hunk 5]
```diff
@@ -78,3 +80,35 @@ def rules_as_rich(rules: RulesCollection) -> Iterable[Table]:
         if rule.severity:
             table.add_row("severity", rule.severity)
         yield table
+
+
+def profiles_as_md() -> str:
+    """Return markdown representation of supported profiles."""
+    result = ""
+    default_rules = RulesCollection([DEFAULT_RULESDIR])
+    # breakpoint()
```
Comment (ganeshrn) [inline src/ansiblelint/generate_docs.py:89]: Is this comment required? If not it can be removed.

**Review by ganeshrn — APPROVED** [review]
Good to merge from my side after the minor comments are fixed

File: src/ansiblelint/__main__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cognifloyd) [reply src/ansiblelint/__main__.py:181]: Got it. I'm glad it's doing something sane. I'm not a fan of argparse, but it's not worth replacing it at this point.

**Review by priyamsahoo — APPROVED** [review]

**Pull request merged**